mindsdb/integrations/handlers/huggingface_handler/huggingface_handler.py

```diff
@@ -346,38 +347,143 @@ def describe(self, attribute: Optional[str] = None) -> pd.DataFrame:
             tables = ["args", "metadata"]
             return pd.DataFrame(tables, columns=["tables"])
 
-    def finetune(
-        self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None
-    ) -> None:
-        finetune_args = args if args else {}
-        args = self.base_model_storage.json_get("args")
-        args.update(finetune_args)
+    def finetune(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        logger.info("Starting finetune method")
+        self.print_debug_info()
+
+        if df is None or df.empty:
+            raise ValueError("No data provided for fine-tuning")
+
+        logger.info(f"Input DataFrame - shape: {df.shape}, columns: {df.columns}")
+
+        # Retrieve stored args, defaulting to an empty dict if None
+        stored_args = self.model_storage.json_get("args") or {}
+        
+        # Merge stored args with provided args, prioritizing provided args
+        finetune_args = {**stored_args, **(args or {})}
+        
+        # Check if 'using' key exists and extract its contents
+        if 'using' in finetune_args:
+            finetune_args.update(finetune_args['using'])
+            del finetune_args['using']
+
+        logger.debug(
+            f"Arguments:\n"
+            f"  Stored args: {stored_args}\n"
+            f"  Combined finetune args: {finetune_args}"
+        )
+
+        model_name = finetune_args.get("model_name")
+        if not model_name:
+            raise ValueError("Model name not found in arguments. Please ensure the model was created correctly.")
 
-        model_name = args["model_name"]
         model_folder = self.model_storage.folder_get(model_name)
-        args["model_folder"] = model_folder
+        finetune_args["model_folder"] = model_folder
         model_folder_name = model_folder.split("/")[-1]
-        task = args["task"]
+        task = finetune_args.get("task")
+        if not task:
+            raise ValueError("Task not specified in arguments")
 
         if task not in FINETUNE_MAP:
             raise KeyError(
                 f"{task} is not currently supported, please choose a supported task - {', '.join(FINETUNE_MAP)}"
             )
 
-        tokenizer, trainer = FINETUNE_MAP[task](df, args)
+        # Ensure the input column and target column are correctly specified
+        input_column = finetune_args.get("input_column")
```

Comment:
What is the need for an 'input_column' here? Shouldn't it be possible to finetune the model on many inputs?